File: testing-kit/templates/unit.test.template.ts
// Copy this file into your project's tests/ folder, rename, edit imports.
// Pattern: pure function unit tests grouped by "normal" / "hard" cases.
//
// Run: npm test

import { describe, it, expect } from 'vitest';
// import { yourFunction } from '@/lib/your-module';

describe('your-module — normal cases', () => {
  it('does the happy path', () => {
    // expect(yourFunction(1, 2)).toBe(3);
    expect(1 + 2).toBe(3);
  });

  it('handles a typical input', () => {
    expect(true).toBe(true);
  });
});

describe('your-module — hard / edge cases', () => {
  it('handles invalid input gracefully', () => {
    // expect(() => yourFunction(null as any)).not.toThrow();
    expect(() => null).not.toThrow();
  });

  it('returns the expected value at the boundary', () => {
    expect([0, -1, Number.MAX_SAFE_INTEGER]).toContain(0);
  });
});

File: testing-kit/templates/unit.test.template.py
"""Pure-function pytest template — copy into tests/, rename, edit imports.

Run: pytest
"""
import pytest

# from app.your_module import your_function


class TestYourModuleNormal:
    def test_happy_path(self):
        # assert your_function(1, 2) == 3
        assert 1 + 2 == 3

    def test_typical_input(self):
        assert True


class TestYourModuleEdge:
    def test_invalid_input_does_not_raise(self):
        # Replace with the call you want to guard.
        try:
            None
        except Exception:
            pytest.fail("should not raise")

    def test_boundary_value(self):
        assert 0 in [0, -1, 2**63 - 1]
